portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -14,78 +14,92 @@
 
 package com.google.sps.servlets;
 
-import com.google.appengine.api.datastore.DatastoreService;
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
-import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.*;
 import com.google.gson.Gson;
+import com.google.sps.utils.CommentDataStore;
 import com.google.sps.data.Comment;
+import com.google.sps.utils.CommentUtils;
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
 
 /** Servlet that handles getting and posting comment content. */
 @WebServlet("/comment")
 public final class CommentServlet extends HttpServlet {
-  private final List<Comment> comments = new ArrayList<>();
+
+  static class Constants {
+    final static String COMMENT_COMMENTER = "commenter";
+    final static String COMMENT_CONTENT = "content";
+  }
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(this.comments);
+    // Loads the comment from Datastore
+    List<Comment> comments = getComments();
 
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+    // Converts into json form
+    String json = convertToJsonUsingGson(comments);
+
+    // Sends the JSON as the response
+    sendJsonResponse(response, json);
   }
 
+  /** Loads the comment from Datastore */
+  private List<Comment> getComments() {
```

Comment:
Maybe this conversion part can be in the datastore itself. It returns a list of comments to you.

Reply:
Fixed